pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -209,6 +216,14 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	// record an event about the resources we are now attempting to reclaim via eviction
 	m.recorder.Eventf(m.nodeRef, api.EventTypeWarning, "EvictionThresholdMet", "Attempting to reclaim %s", resourceToReclaim)
 
+	// check if there are node-level resources we can reclaim to reduce pressure before evicting end-user pods.
+	if m.reclaimNodeLevelResources(resourceToReclaim, observations) {
```

Comment:
i see your point now after further reflection.

in the case of pod eviction, we score pods by a primary resource, and we prioritize reclaiming memory over disk.  for node-level reclaim, i could see just reclaiming all starved resources.  in practice right now, we have no memory based reclaim.  for disk, i guess if both nodefs and imagefs were starved, we would get a benefit in calling the super set of functions.